test/functional/p2p_segwit.py

```diff
@@ -1999,21 +1998,15 @@ class msg_bogus_tx(msg_tx):
             def serialize(self):
                 return serialize_with_bogus_witness(self.tx)
 
-        self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(address_type='bech32'), 5)
-        self.generate(self.nodes[0], 1)
-        unspent = next(u for u in self.nodes[0].listunspent() if u['spendable'] and u['address'].startswith('bcrt'))
-
-        raw = self.nodes[0].createrawtransaction([{"txid": unspent['txid'], "vout": unspent['vout']}], {self.nodes[0].getnewaddress(): 1})
-        tx = tx_from_hex(raw)
+        wallet = MiniWallet(self.nodes[0])
+        self.generate(wallet, COINBASE_MATURITY + 1)
```

Comment:
nit: This can be skipped if you mine the 99 first blocks with miniwallet instead.

```py
        self.generate(self.nodes[0], 99)  # let the block mature

Reply:
Thanks, done. The MiniWallet instance is now also created in `run_test` before any sub-tests are executed, which makes sense. As potential follow-up idea, I guess there are also some refactoring possibilities to use MiniWallet rather than manually crafting CTransaction instances in other sub-tests accessing `self.utxo` (didn't check in detail yet though).